durable-storage/src/merkle_worker.rs

```diff
@@ -75,7 +76,8 @@ impl MerkleWorker {
         async_handle: &Handle,
         persistence_layer: Arc<PersistenceLayer>,
     ) -> Result<Self, MerkleWorkerError> {
-        let layer = MerkleLayer::new(persistence_layer);
+        let node_resolver = Arc::new(FullMavlNodeResolver::new(persistence_layer.clone()));
+        let layer = MerkleLayer::<FullMavlNodeResolver>::new(persistence_layer, node_resolver);
```

Comment:
```suggestion
        let layer = MerkleLayer::new(persistence_layer, node_resolver);
```
this type annotation is superfluous